### PR TITLE
Upgrade to Gradle 8

### PR DESCRIPTION
**Pull request by m0ne**
- Gradle 6 -> 8
- Java 8 -> 11
- Upgrade of the dependencies

### CODE REVIEW

File: crystal-map-api/build.gradle [hunk 1]
```diff
@@ -1,20 +1,21 @@
 apply plugin: 'kotlin'
-apply plugin: 'maven'
+apply plugin: 'maven-publish'
```
Comment (m0ne) [inline crystal-map-api/build.gradle:2]: 'maven' does no longer exist.

File: crystal-map-api/src/main/java/com/schwarz/crystalapi/util/CrystalWrap.kt [hunk 2]
```diff
@@ -1,3 +1,5 @@
+@file:Suppress("UNCHECKED_CAST")
```
Comment (m0ne) [inline crystal-map-api/src/main/java/com/schwarz/crystalapi/util/CrystalWrap.kt:1]: Can be removed, but surpresses most of the warnings in CrystalWrap

File: crystal-map-api/src/main/java/com/schwarz/crystalapi/util/MapifyUtil.kt [hunk 3]
```diff
@@ -1,3 +1,5 @@
+@file:Suppress("UNCHECKED_CAST")
```
Comment (m0ne) [inline crystal-map-api/src/main/java/com/schwarz/crystalapi/util/MapifyUtil.kt:1]: Same as above

File: crystal-map-couchbase-connector/build.gradle [hunk 4]
```diff
@@ -21,30 +21,35 @@ android {
...
 }
 
+sourceCompatibility = JavaVersion.VERSION_11
+targetCompatibility = JavaVersion.VERSION_11
+
 group = 'com.github.SchwarzIT'
 
-task sourcesJar(type: Jar) {
-    classifier = 'sources'
+tasks.register('sourcesJar', Jar) {
```
Comment (m0ne) [inline crystal-map-couchbase-connector/build.gradle:36]: new way of defining tasks

File: crystal-map-couchbase-connector/build.gradle [hunk 5]
```diff
@@ -1,12 +1,12 @@
 apply plugin: 'com.android.library'
-apply plugin: 'com.github.dcendents.android-maven'
 apply plugin: 'kotlin-android'
 
 android {
-    compileSdkVersion 33
+    namespace = "com.schwarz.cystalcouchbaseconnector"
+    compileSdk 33
```
Comment (m0ne) [inline crystal-map-couchbase-connector/build.gradle:6]: compileSdkVersion deprecated

File: crystal-map-processor/src/main/java/com/schwarz/crystalprocessor/model/accessor/CblGenerateAccessorHolder.kt [hunk 6]
```diff
@@ -14,24 +14,26 @@ class CblGenerateAccessorHolder(
...
                 methodBuilder.addParameter(it.name, it.type)
             }
 
-            methodBuilder.addStatement(
-                "return %T.%N(${callParams.joinToString()})" + System.lineSeparator(),
-                sourceClassTypeName,
-                memberFunction.name
-            )
+            methodBuilder
+                .returns(memberFunction.returnTypeName)
```
Comment (m0ne) [inline crystal-map-processor/src/main/java/com/schwarz/crystalprocessor/model/accessor/CblGenerateAccessorHolder.kt:31]: always expected "Unit" as the return type, should now specify the proper return type

File: crystal-map-versioning-plugin/src/main/java/com/schwarz/crystalversioningplugin/task/ValidationTask.kt [hunk 7]
```diff
@@ -25,7 +28,7 @@ open class ValidationTask : DefaultTask() {
             val currentVersionFile = parseVersionSchema(File(extension.currentSchema))
             val prettyPrinter = services.get(StyledTextOutputFactory::class.java)
 
-            val validator = it.newInstance()
+            val validator = it.getDeclaredConstructor().newInstance()
```
Comment (m0ne) [inline crystal-map-versioning-plugin/src/main/java/com/schwarz/crystalversioningplugin/task/ValidationTask.kt:31]: auto correct from intellij

File: demo/src/main/AndroidManifest.xml [hunk 8]
```diff
@@ -1,6 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="com.schwarz.crystaldemo">
```
Comment (m0ne) [inline demo/src/main/AndroidManifest.xml:3]: package in manifest is deprecated

**Review by m0ne — COMMENTED** [review]
One run of ktlint format, @sbra0902 are the empty lines after the function definitions desired? I could modify the ktlint rules if this is intended

File: crystal-map-api/src/main/java/com/schwarz/crystalapi/schema/DefaultSchemaValidator.kt [hunk 9]
```diff
@@ -16,9 +16,7 @@ open class DefaultSchemaValidator : SchemaValidator {
     }
 
     protected open fun validateModelLevel(current: EntitySchema?, released: EntitySchema, logger: SchemaValidationLogger): Boolean {
-
```
Comment (m0ne) [inline crystal-map-api/src/main/java/com/schwarz/crystalapi/schema/DefaultSchemaValidator.kt:19]: One run of ktlint format, @sbra0902 are the empty lines after the function definitions desired? I could modify the ktlint rules if this is intended

**Comment by sbra0902** [comment]
obsolet

**Pull request closed without merging**